tests/jballerina-unit-test/src/test/resources/test-src/types/xml/xml-literals-negative.bal

```diff
@@ -72,3 +72,16 @@ function testXmlNsInterpolation() returns xml {
     xml x = xml `<foo xmlns="${ns}" xmlns:foo="${ns}">hello</foo>`;
     return x;
 }
+
+function testXMLLiteralWithEscapeSequence(){
+    xml x1 = xml `hello &lt; &gt; &amp;`;
+    int i = 0;
+    string[] strs = [];
+    foreach string|xml e in x1 {
+        if e is string {
+            strs[i] = e;
+            i += 1;
+        }
+    }
```

Comment:
since we changed the iterator return type to xml fom xml|string this tests if we get the below error.

>  incompatible types: 'xml' will not be matched to 'string'

Reply:
Sorry, missed that this was a negative test. In the long term though, this will become outdated once we allow the type test with intersections. Let's leave this as is until then.